frappe/model/db_query.py

```diff
@@ -128,6 +128,20 @@ def build_and_run(self):
 			args.fields = 'distinct ' + args.fields
 			args.order_by = '' # TODO: recheck for alternative
 
+		if args.order_by and args.group_by:
```

Comment:
What does this block aim to do? Can we have some comments added here?

Reply:
Postgres requires anything that appears in the select fields to also appear in the order by and group by statements. MariaDB does not. This addresses that issue. I can add a comment to that effect.

Reply:
Comment added